### PR TITLE
Port foreman_job_template to new foreman_helper

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: plugins/modules/foreman_job_template.py [hunk 1]
```diff
@@ -249,95 +249,88 @@
...
-        JobTemplate,
-        TemplateInput,
-        find_entities,
-        find_entities_by_name,
-        find_template_input,
-        ForemanEntityAnsibleModule,
-        naildown_entity,
-        naildown_entity_state,
-        sanitize_entity_dict,
-    )
```
Comment (evgeni) [inline plugins/modules/foreman_job_template.py]: Where is `os` used? I can't find it in neither the old nor the new code

File: plugins/modules/foreman_job_template.py [hunk 2]
```diff
@@ -249,95 +249,88 @@
...
 }
 
 
 def main():
     module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
-            # Entity parameter
+        entity_spec=dict(
             audit_comment=dict(),
             description_format=dict(),
```
Comment (evgeni) [inline plugins/modules/foreman_job_template.py]: Can we remove this effective user line?

File: plugins/modules/foreman_job_template.py (hunk 1 above)
Comment (mdellweg) [reply plugins/modules/foreman_job_template.py]: there is some os.path manipulation around line 374.

Comment (evgeni) [reply plugins/modules/foreman_job_template.py]: Ooh, I've been fooled by the diff view, didn't see that there were collapsed lines. Or rather not used to see that in a nailgun to apypie PR ;)

File: plugins/modules/foreman_job_template.py (hunk 2 above)
Comment (mdellweg) [reply plugins/modules/foreman_job_template.py]: sure. A quick look at the api shows, that its implementation is at least not that easy.

File: plugins/modules/foreman_job_template.py (hunk 1 above)
Comment (mdellweg) [reply plugins/modules/foreman_job_template.py]: Maybe now it not the best time anymore to _get used_ to seeing nailgun-to-apypie-prs.

File: plugins/modules/foreman_job_template.py [hunk 3]
```diff
@@ -388,72 +380,66 @@ def main():
     affects_multiple = name == '*'
     # sanitize user input, filter unuseful configuration combinations with 'name: *'
     if affects_multiple:
-        if state == 'present_with_defaults':
+        if module.state == 'present_with_defaults':
             module.fail_json(msg="'state: present_with_defaults' and 'name: *' cannot be used together")
-        if state == 'absent':
+        if module.state == 'absent':
```
Comment (evgeni) [inline plugins/modules/foreman_job_template.py]:
```suggestion
        if module.desired_absent:
```

Even tho I am not sure the check is *really* useful ;)

**Pull request merged**